census_view.cpp

```diff
@@ -1268,8 +1268,44 @@ void CensusView::apply_changes
         }
 }
 
-void CensusView::update_visible_columns()
+bool CensusView::update_visible_columns()
 {
+    // Check if we really need to update the columns, it takes a relatively
+    // long time and is done directly in response to the user actions, so it
+    // makes a lot of sense to avoid doing it completely as long as possible.
+    bool must_update_columns = false;
+
+    auto const& all_headers(case_parms()[0].member_names());
+    if(all_headers != all_headers_)
+        {
+        // The entire set of columns has changed, we have no choice but to
+        // reexamine all of them to check which ones should be shown.
+        all_headers_ = all_headers;
+        headers_show_.resize(all_headers_.size(), false);
+        must_update_columns = true;
+        }
+
```

Comment:
I realize this is my own code, but I think there should be `else` here as it doesn't make sense to do anything else if `must_update_columns` is already `true`.

Reply:
> 
> 
> I realize this is my own code, but I think there should be `else` here as it doesn't make sense to do anything else if `must_update_columns` is already `true`.

Actually it's even worse than this: if `all_headers != all_headers_` there doesn't seem to be any reasons for `headers_show` to have as many elements as `all_headers`, so we could just crash below...